public/chirpy.js

```diff
@@ -345,12 +393,30 @@ function checkInputAndHighlightButtons() {
             option.select('input')
                 .property('checked', true);
             option.classed('selectedLabel', true);
+            numberMatched++;
         } else {
             option.select('input')
                 .property('checked', false);
             option.classed('selectedLabel', false);
         }
     });
+    subStrings.forEach(function(subString) {
+        if(subString.length === 0) {
+            numberMatched++;
+        }
+    });
+
+    var nonWhiteSpace = userInputText.search( /\S/ );
```

Comment:
Hi Lawrie, please can you move the variable declaration to the top of the function?   Thanks.